src/algebra/ring.lean

```diff
@@ -194,28 +279,117 @@ end is_ring_hom
 
 set_option old_structure_cmd true
 
+/-- Bundled semiring homomorphisms; use this for bundled ring homomorphisms too. -/
+structure semiring_hom (α : Type*) (β : Type*) [semiring α] [semiring β]
+  extends monoid_hom α β, add_monoid_hom α β
+
+infixr ` →+* `:25 := semiring_hom
+
+instance {α : Type*} {β : Type*} [semiring α] [semiring β] : has_coe_to_fun (α →+* β) :=
+⟨_, semiring_hom.to_fun⟩
+
+namespace semiring_hom
+
+variables {β : Type v} [semiring α] [semiring β]
+variables (f : α →+* β) {x y : α}
+
+@[extensionality] theorem ext (f g : α →+* β) (h : (f : α → β) = g) : f = g :=
+by cases f; cases g; cases h; refl
+
+/-- Semiring homomorphisms map zero to zero. -/
+@[simp] lemma map_zero (f : α →+* β) : f 0 = 0 := f.map_zero'
+
+/-- Semiring homomorphisms map one to one. -/
+@[simp] lemma map_one (f : α →+* β) : f 1 = 1 := f.map_one'
+
+/-- Semiring homomorphisms commute with addition. -/
+@[simp] lemma map_add (f : α →+* β) (a b : α) : f (a + b) = f a + f b := f.map_add' a b
+
+/-- Semiring homomorphisms commute with multiplication. -/
+@[simp] lemma map_mul (f : α →+* β) (a b : α) : f (a * b) = f a * f b := f.map_mul' a b
+
+instance {α : Type*} {β : Type*} [semiring α] [semiring β] (f : α →+* β) :
+  is_semiring_hom (f : α → β) :=
+{ map_zero := map_zero f,
+  map_one := map_one f,
+  map_add := map_add f,
+  map_mul := map_mul f   }
```

Comment:
```suggestion
  map_mul := map_mul f }
```
Also, can we have a similar instance for `ring`s and `is_ring_hom`?

Reply:
Will the instance on line 347 do or should I have directly provided the structure fields instead of using ```is_ring_hom.of_semiring```? (And should it be on line 318?)

Reply:
Oh, sorry, I missed that one. I guess I was looking for a longer block of code, with structure fields. No, this is totally fine.